### ISSUE TITLE
Do not report "use before definition" errors for .d.ts files

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#31746]
**TypeScript Version:**  3.6.0-dev.20190603

**Search Terms:** use before definition declaration

**Code**

```ts
// foo.d.ts
export declare class C {
  [x]: string;
}
export declare const x: unique symbol;
```

**Expected behavior:**

No error.

**Actual behavior:**

```
(TS2729) Property 'x' is used before its initialization.
```

Although a "use before definition"-type error makes sense in a .ts file where evaluation order matters, a .d.ts file isn't itself evaluated so there is no reason to report an error. 

[This was discovered](https://github.com/microsoft/web-build-tools/pull/1305#discussion_r289978662) when using the dts rollup feature of `@microsoft/api-extractor`.

**Comment by andrewbranch** [comment]
Hey @rbuckton, I’m having trouble reproducing this, both in a real d.ts file and in the compiler tests. I have all compiler options strict. Is there anything else I need to do to see this behavior?

**Comment by rbuckton** [comment]
This repros:
1. In an empty directory:
    1. Add a tsconfig.json with the following content:
    ```json
    {
      "compilerOptions": {
        "target": "es2015",
        "module": "commonjs"
      }
    }
    ```
    1. Add a test.d.ts with the following content:
    ```ts
    export declare class ClassWithSymbols {
      public readonly [Namespace.locallyExportedCustomSymbol]: string;
      public [Namespace.fullyExportedCustomSymbol](): void;
    }
    export namespace Namespace {
      export const locallyExportedCustomSymbol: unique symbol;
      export const fullyExportedCustomSymbol: unique symbol;
    }
    ```

1. Open test.d.ts in vscode, the error occurs on line 2:
    ![image](https://user-images.githubusercontent.com/3902892/61835911-b3398580-ae32-11e9-87b6-7a85f2666618.png)

**Comment by rbuckton** [comment]
Confirmed in 3.5.2 and 3.6.0-beta

**Comment by rbuckton** [comment]
My apollogies, my minimal repro was apparently too minimal and didn't trigger the condition actually encountered in https://github.com/microsoft/web-build-tools/pull/1305#discussion_r289978662
